packit_service/worker/github_handlers.py

```diff
@@ -0,0 +1,304 @@
+            repo_namespace = event.base_repo_namespace
+            base_ref = event.base_ref
+        elif isinstance(event, ReleaseEvent):
+            repo_name = event.repo_name
+            repo_namespace = event.repo_namespace
+            base_ref = event.tag_name
+
+        self.project: GitProject = get_github_project(
+            self.config, repo=repo_name, namespace=repo_namespace
+        )
+        self.package_config: PackageConfig = get_package_config_from_repo(
+            self.project, base_ref
+        )
+        self.package_config.upstream_project_url = event.https_url
+
+    def handle_pull_request(self):
+        if not self.job.metadata.get("targets"):
+            logger.error(
+                "'targets' value is required in packit config for copr_build job"
+            )
+
+        self.local_project = LocalProject(
+            git_project=self.project,
+            pr_id=self.event.pr_id,
+            git_service=self.project.service,
+            working_dir=self.config.command_handler_work_dir,
+        )
+        self.api = PackitAPI(self.config, self.package_config, self.local_project)
+
+        default_project_name = (
+            f"{self.project.namespace}-{self.project.repo}-{self.event.pr_id}"
+        )
+        owner = self.job.metadata.get("owner") or "packit"
+        project = self.job.metadata.get("project") or default_project_name
+        r = BuildStatusReporter(self.project, self.event.commit_sha)
+
+        try:
+            build_id, repo_url = self.api.run_copr_build(
+                owner=owner, project=project, chroots=self.job.metadata.get("targets")
+            )
+        except SandcastleTimeoutReached:
+            msg = "You have reached 10-minute timeout while creating the SRPM."
+            self.project.pr_comment(self.event.pr_id, msg)
+            msg = "Timeout reached while creating a SRPM."
+            r.report("failure", msg)
+            return HandlerResults(success=False, details={"msg": msg})
+        except SandcastleCommandFailed as ex:
+            max_log_size = 1024 * 16  # is 16KB enough?
+            if len(ex.output) > max_log_size:
+                output = "Earlier output was truncated\n\n" + ex.output[-max_log_size:]
+            else:
+                output = ex.output
+            msg = (
+                "There was an error while creating a SRPM.\n"
+                "\nOutput:"
+                "\n```\n"
+                f"{output}"
+                "\n```"
+                f"\nReturn code: {ex.rc}"
+            )
+            self.project.pr_comment(self.event.pr_id, msg)
+            msg = "Failed to create a SRPM."
+            r.report("failure", msg)
+            return HandlerResults(success=False, details={"msg": msg})
+        except FailedCreateSRPM:
+            msg = "Failed to create a SRPM."
+            r.report("failure", msg)
+            return HandlerResults(success=False, details={"msg": msg})
```

Comment:
My fix is missing here: https://github.com/packit-service/packit-service/commit/9dbb371f1e6f3f72420ddea9f09f8c9a0655c47d

Reply:
good catch! should be fixed now